src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1053,17 +1054,26 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 	WorkerNode *workerNode = ModifiableWorkerNode(nodeName, nodePort);
 	if (NodeIsPrimary(workerNode))
 	{
+		if (CanRemoveReferenceTablePlacements())
+		{
+			/*
+			 * Delete reference table placements so they are not taken into account
+			 * for the check if there are placements after this.
+			 */
+			DeleteAllReferenceTablePlacementsFromNodeGroup(workerNode->groupId);
+		}
 		bool onlyConsiderActivePlacements = false;
-
-		/*
-		 * Delete reference table placements so they are not taken into account
-		 * for the check if there are placements after this
-		 */
-		DeleteAllReferenceTablePlacementsFromNodeGroup(workerNode->groupId);
-
 		if (NodeGroupHasShardPlacements(workerNode->groupId,
 										onlyConsiderActivePlacements))
 		{
+			if (ClusterHasReferenceTable())
+			{
+				ereport(ERROR, (errmsg(
+									"you cannot remove node because there are reference "
+									"tables and you would lose the data on reference tables"),
```

Comment:
I think we do not prefer "you" word in error messages + emphasizing that it is the "last" node::

```suggestion
									"cannot remove the last worker node because there are reference "
									"tables and it would cause data loss for reference tables"),
```

Reply:
I tried to make it similar with the existent one. I guess we can change both of them or none of them.